### PR TITLE
Add val/var modifiers to primary constructor params, fixes #136

### LINKED ISSUE #136: Disappearing val/var modifiers in class parameters
Reposted from https://github.com/scalameta/scalameta/issues/555

```
class Identity extends scala.annotation.StaticAnnotation {
  inline def apply(defn: Any): Any = meta {  defn }
}

@Identity
class Test(val value:Int) extends AnyVal
```

Doesn't work, because the `val` modifier is lost and then the `AnyVal` infrastructure complains about that.

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: tests/converter/src/test/scala/org/scalameta/tests/LotsOfProjects.scala [hunk 1]
```diff
@@ -70,6 +70,6 @@ object LotsOfProjects extends ConverterSuite {
   def main(args: Array[String]): Unit = {
     val results = getResults
     printResults(results)
-    assert(results.count(_ == "Success") >= 25951) // increment this number as it increases.
+    assert(results.count(_ == "Success") >= 25950) // increment this number as it increases.
```
Comment (olafurpg) [inline tests/converter/src/test/scala/org/scalameta/tests/LotsOfProjects.scala]: The decrease is caused by this file: https://github.com/scala/scala/blob/ad48e5918081c679546f50b6f52dd8e0813754e7/test/files/run/t9437b/Foo_1.scala

There is a bug in how `displayName` handles the

```
`_`
```
identifier. I don't see a reason to fix that bug in this PR.

File: tests/converter/src/test/scala/org/scalameta/tests/Syntactic.scala [hunk 2]
```diff
@@ -121,6 +121,14 @@ class Syntactic extends ConverterSuite {
   // syntactic("val (c, matSink: (Int, String)) = (a, b)")
 
   // definitions
+  syntactic("case class X(val x: Int)")
+  syntactic("case class X(x: Int)")
```
Comment (xeno-by) [inline tests/converter/src/test/scala/org/scalameta/tests/Syntactic.scala]: What about `case class X(private[this] x: Int)`?

File: tests/converter/src/test/scala/org/scalameta/tests/LotsOfProjects.scala (hunk 1 above)
Comment (xeno-by) [inline tests/converter/src/test/scala/org/scalameta/tests/LotsOfProjects.scala]: Allright np.

File: tests/converter/src/test/scala/org/scalameta/tests/Syntactic.scala (hunk 2 above)
Comment (olafurpg) [inline tests/converter/src/test/scala/org/scalameta/tests/Syntactic.scala]: Done. Did you expect it to be different from the `class X(private[this] x: Int)` test below?

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**